fetch.js

```diff
@@ -443,6 +468,10 @@
         reject(new TypeError('Network request failed'))
       }
 
+      xhr.onabort = function() {
```

Comment:
@mislav I believe this is the most correct as mentioned [here](https://github.com/github/fetch/pull/572#issuecomment-337740989). Calling abort on the XHR will set the right status code and `readyState` (https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/abort).

Reply:
I'm fine with `xhr.abort()`. I'm just saying that we don't need to install the `xhr.onabort` _callback_ just to reject the promise. We know that `abortXhr()` will be executed, so we can reject the promise in there.

Reply:
I see.. xhr is never exposed so it can't be aborted from the outside. Is there any other way it could be aborted by the browser? If not, there shouldn't be any difference. I don't mind this way though, reads well :)

Reply:
`onabort` wasn't needed until now, and since we control literally the only place from which the xhr is aborted, I don't think adding one level of indirection on top of `abortXhr` is necessary.

Reply:
Doesn't `onAbort()` also confirm that `xhr.abort()` was successful vs. just trusting the results of `abortXhr()`?

Reply:
@dysonpro That's correct, but we aren't interested in whether `xhr.abort()` was performed successfully or not. We want to reject the promise right away because we know for sure an abort was requested.

Reply:
Thanks @mislav for your explanation. I am excited about this functionality so I have been following this PR carefully. I am already using this branch code in a project that requires aborting no longer needed requests.
Your point is that you want to ensure the promise is rejected, correct? I feel that it is closer to the core intention of the PR (and my own project as well) to ensure that the abort was completed successfully. Is there a way to do that without utilizing the `xhr.onAbort()` callback? Without that I feel that there could be a gap in the verification of the functionality for every platform and environment. Thanks again for your feedback.

Reply:
@dysonpro Your concerns are valid, but we have different priorities when thinking about this. At the point when `abortXhr()` is invoked (which can only happen upon AbortSignal), I consider the request most definitely discarded. We can then call `xhr.abort(); reject(...)` all at once. Now, if there is a browser which would somehow fail to actually abort the XHR when `xhr.abort()` is called, that is fine with me because we already ensured that the promise is rejected. I don't expect this case to happen often (or ever, for that matter) and so I'm not concerned with it. The code change that I suggested, after all, is mostly stylistic.

Reply:
> Is there any other way it could be aborted (e.g. by the browser)? If not, there shouldn't be any difference.

* A browser extension (e.g. an ad blocker) could abort a request.
* A service worker could intercept the fetch and replace it with a new request that has (or will have) an aborted signal.

Arguably, these are rather obscure use cases, but they're real nonetheless. I think a general `fetch` polyfill should handle these, so it should listen for `xhr.onabort`.